sdk/servicebus/service-bus/src/receivers/sessionReceiver.ts

```diff
@@ -570,21 +569,18 @@ export class SessionReceiverImpl<ReceivedMessageT extends ReceivedMessage | Rece
       throw new TypeError("The parameter 'onError' must be of type 'function'.");
     }
 
-    this._createMessageSessionIfDoesntExist()
-      .then(async () => {
-        if (!this._messageSession) {
-          return;
-        }
-        if (!this._isClosed) {
-          this._messageSession.receive(onMessage, onError, options);
-        } else {
-          await this._messageSession.close();
-        }
-        return;
-      })
-      .catch((err) => {
-        onError(err);
-      });
+    try {
+      if (!this._isClosed) {
+        this._messageSession.receive(onMessage, onError, options);
+      } else {
+        this._messageSession.close().catch((err) => {
```

Comment:
What was the reason to change from await to catch?

Reply:
Since I've removed the `this._createMessageSessionIfDoesntExist()`, there was no scope to pass this block as a callback. Then, I felt there was no need for this block to be wrapped as a function, removed the arrow function, and lost privileges to await.

I can make it an async iife and retain the await.

Reply:
Updated.